frontend/src/js/preview/actions.ts

```diff
@@ -24,9 +24,9 @@ export function useOpenPreview() {
     dispatch(loadCSVForPreview.request());
 
     try {
-      const parsed = await loadCSV(url);
+      const result = await loadCSV(url);
 
-      dispatch(loadCSVForPreview.success({ csv: parsed.result.data, columns }));
+      dispatch(loadCSVForPreview.success({ csv: result.data, columns }));
```

Comment:
Not something for this PR, but I'd advise you to look into stream processing (using the step parameter) of the csv (For example if we are going to do a redesign or add features).

It could probably be much more snappy/responsive if you didn't have to store the entire CSV in memory for preview, especially since as far as I know only the first 500 or so lines are actually preivewable. It can sometimes freeze the whole frontend for bigger files

Reply:
Check 👍